apps/dot-voting/app/store/votes.js

```diff
@@ -226,8 +229,76 @@ const marshallVote = ({
     totalParticipation: totalParticipation,
     metadata,
     executionScript,
+    executionTargetData,
     executed,
     participationPct:
       totalVoters === 0 ? 0 : (totalParticipation / totalVoters) * 100,
   }
 }
+
+const getVoteExecutionTargets = (vote) => {
+  return vote.executionTargets || []
+}
+
+const decorateVote = async (vote) => {
+  return new Promise(resolve => {
+    const executionTargets = getVoteExecutionTargets(vote)
+    let targetApp
+    if (!executionTargets.length) {
+      // If there's no execution target, consider it targetting this Dot Voting app
+      app.currentApp()
+        .pipe(first())
+        .subscribe((currentApp) => {
+          resolve({
+            ...vote,
+            executionTargetData: {
+              ...currentApp,
+              identifier: undefined,
+            }
+          })
+        })
+    } else if (executionTargets.length > 1) {
+      // If there's multiple targets, make a "multiple" version
+      resolve({
+        ...vote,
+        executionTargetData: {
+          address: '0x0000000000000000000000000000000000000000',
+          name: 'Multiple',
+          iconSrc: null,
+          identifier: undefined,
+        }
+      })
+    } else {
+      // Otherwise, try to find the target from the installed apps
+      app.installedApps()
+        .pipe(first())
+        .subscribe((installedApps) => {
+          const [targetAddress] = executionTargets
+
+          targetApp = installedApps.find(app => app.appAddress === targetAddress)
+          if (!targetApp) {
+            resolve({
+              ...vote,
+              executionTargetData: {
+                address: targetAddress,
+                name: 'External',
+                iconSrc: null,
+                identifier: undefined,
+              }
+            })
+          } else {
+            const { appAddress, icon, identifier, name } = targetApp
+            resolve({
+              ...vote,
+              executionTargetData: {
+                address: appAddress,
+                name,
+                iconSrc: icon(24),
```

Comment:
`icon` naming seems confusing, it seemed to be an object but then it is a function, and the param passed to it seems so cryptic, I think at least a comment here would help any future maintainer

Reply:
So any idea about why icon is a function? is this the best name we have for that? As a developer, do you honestly think that it will be clear for other developers to understand what is going on with `iconSrc: icon(24)`

Reply:
Ok thanks for the clarifying comment, the icon() function still seems extremely confusing, but at least one can understand what it does after this long explanation.

Reply:
I think something like `getIconWithSize(24)` - for example - would be self-explaining in this case, I let you final judgement on that @PeterMPhillips

Reply:
The function is passed by the Aragon API's installedApps() function. This is an issue we'd have to take up in that repo